modules/core/src/controllers/controller.ts

```diff
@@ -110,14 +110,34 @@ export type InteractionState = {
   isZooming?: boolean;
 }
 
+export type ViewState = {
+  zoom: number;
+  longitude: number;
+  latitude: number;
+  bearing?: number;
+  pitch?: number;
+  transitionDuration?: number;
+  width?: number;
+  height?: number;
+  altitude?: number;
+  maxPitch?: number;
+  maxZoom?: number;
+  minPitch?: number;
+  minZoom?: number;
+  normalize?: any;
+  position?: number[];
+  transitionEasing?: any;
+  transitionInterpolator?: any;
+}
+
 /** Parameters passed to the onViewStateChange callback */
 export type ViewStateChangeParameters = {
   /** The next view state, either from user input or transition */
-  viewState: Record<string, any>;
+  viewState: ViewState;
```

Comment:
This is assuming a `WebMercatorViewport`, we also have other view types, e.g. https://deck.gl/docs/api-reference/core/orthographic-view#view-state

Adding types here is a good idea, but rather than duplicating the type here, it is better to import the types used elsewhere